src/borg/archiver/diff_cmd.py

```diff
@@ -43,17 +43,36 @@ def print_text_output(diff, path):
 
         matcher = build_matcher(args.patterns, args.paths)
 
-        diffs = Archive.compare_archives_iter(
-            archive1, archive2, matcher, can_compare_chunk_ids=can_compare_chunk_ids, content_only=args.content_only
+        diffs_iter = Archive.compare_archives_iter(
+            archive1, archive2, matcher, can_compare_chunk_ids=can_compare_chunk_ids
         )
         # Conversion to string and filtering for diff.equal to save memory if sorting
-        diffs = ((path, diff.changes()) for path, diff in diffs if not diff.equal)
+        diffs_list = [diff for diff in diffs_iter if not diff.equal(args.content_only)]
```

Comment:
is it required that this is a list and not an iterator? guess when used with big archives, the list could consume quite a lot of memory.

Reply:
still todo.